src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1266,6 +1266,12 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
       break;
     }
 
+    r = pre_sync_check_and_open_handles(dir_root, current, prev, &fh);
```

Comment:
How is this fixing the bad file descriptor issue? Could you please explain @joscollin.

AFAIK, `fh` doesn't get invalidated till do_synchronize() finishes.

Reply:
It gets invalidated sometimes and it's mostly appear during an `incremental copy based on remote dir_root`. When the copy operation decides to switch the source, then `fh` gets invalidated.  

In this case, It's a minor error with respect to the issue this PR is fixing. The `Bad file descriptor` doesn't appear in the logs when we validate the `fh` every time.
https://pulpito.ceph.com/jcollin-2024-03-27_00:57:31-fs:mirror-wip-jcollin-testing-26032024-distro-default-smithi/